doc/roadmap-2024.md

```diff
@@ -0,0 +1,75 @@
+
+### Data Warmup before Leader Transfer
+
+When the leader of a data range transferred to another TiKV node, because the data of this range is not in memory yet in the new TiKV node, if there are some read queries route to this new leader, it tends to cause higher read latency for these queries. Scenarios like leader re-balancing, TiKV nodes rolling restart may cause leader transfer. We probably need a data warm-up mechanism for the new leader before transfer the leader role to it to reduce the read latency in such scenario.
+
+### Stable Latency During Planned Operations
+
+As a distributed system, there are some planned operations may happen, like rolling restart, rolling upgrade, scale-in some TiKV nodes, scale-out some TiKV nodes, etc. As the cornerstone of other mission critical databases like TiDB, TiKV should provide continuous service and stable latency even when there is a planned operation.
+
+### Stable Latency Under High Data Density Scenario
+
+Some customers' data volumne of single TiKV cluster reach as large as 500TB, and each TiKV node carries more than 4TB data. These large clusters experienced some latency jitter issues, those issues caused by reasons like too many regions in a single TiKV node, background jobs cause resrouces competition, large compaction jobs, impact of large RocksDB instance's global mutex, etc. We are going to elinimate or mitigate these latency jitter issues, and provide stable latency in such high data density scenario.
+
+## Comprehensive Memory Management
+
+### Component-level Memory Trace
+
+Introduce better memory trace for different internal components like transaction module, read pool, raftstore, etc. In this way, we can monitor the memory usage of different components.
+
+### Memory Quota Limit
+
+TiKV has some soft limits for different components' memory usage like scheduler, raftstore, but there are still some components like coprocessor don't have memory quota limit. In the coprocessor, we use the amount of pending queries to throttle the resources usage of coprocessor, but this still can result in Out-Of-Memory(OOM) when there are too many "BIG" queries which retrieve a large amount of data.
+
+## Powerful and Robust Storage Engine
+
+### RocksDB Engine
+
+Newer version RocksDB introduced some promising features like "WAL compression support" and "async-io", we are going to upgrade TiKV's RocksDB to 8.1 version and leverage these powerful features.
+
+### Titan Engine
+
+Titan engine can reduce write amplification dramatically for large value sceanrios, but there are also some drawbacks of Titan engine like space amplification caused by lazy GC, range scan speed has regression in some cases. We will introduce "Punch Hole" feature into Titan to reduce the space amplification. We probably will introduce "Level-Merge" feature to gain better range scan performance for Titan engine.
```

Comment:
We may also do some optimizations to improve the cache efficiency, like prioritize the caching of blob file meta block